engine/classes/Elgg/Http/ResponseFactory.php

```diff
@@ -144,7 +144,10 @@ public function send(Response $response) {
 						. (string) $this->response_sent);
 			}
 		} else {
-			$response->send();
+			if (elgg_trigger_before_event('send', 'http_response', $response)) {
```

Comment:
Aside from PHPUnit applicability, my use case is here: https://github.com/hypeJunction/elgg-cli/blob/master/src/Elgg/CLI/Command.php#L38.
I envision, this will come in handy when working with web services - ultimately, we could allow calling web services by route and having control over how the data is returned will be important.